autogluon/utils/tabular/ml/models/knn/knn_model.py

```diff
@@ -4,18 +4,18 @@
 import time
 
 import psutil
-from sklearn.neighbors import KNeighborsClassifier, KNeighborsRegressor
+from .knn_utils import KNeighborsClassifier, KNeighborsRegressor
 
 from ..abstract import model_trial
-from ..abstract.abstract_model import SKLearnModel
+from ..abstract.abstract_model import AbstractModel
 from ...constants import REGRESSION
 from ....utils.exceptions import NotEnoughMemoryError
 
 logger = logging.getLogger(__name__)
 
 
 # TODO: Normalize data!
-class KNNModel(SKLearnModel):
+class KNNModel(AbstractModel):
```

Comment:
In the first version of this new FAISS implementation, could you instead extend the KNN model to a new model class so we aren't removing the existing KNNModel? Something like FAISSModel? This will help significantly with apples to apples comparisons.

In regards to changing KNNModel to implement AbstractModel instead of SKLearnModel, I believe this is fine as KNN no longer utilizes SKLearnModel to my knowledge.

Reply:
good suggestion - done

Reply:
Perhaps I wasn't quite clear in my first comment. I was referring to the KNNModel class, such as creating a new:

```
class FAISSModel(KNNModel):
    def _get_model_type(self):
        if self.problem_type == REGRESSION:
            return FAISSNeighborsRegressor
        else:
            return FAISSNeighborsClassifier

    def _set_default_params(self):
        default_params = {
            'index_factory_string': 'Flat',
        }
        for param, val in default_params.items():
            self._set_default_param_value(param, val)
        super()._set_default_params()
```

and changing KNNModel slightly to look like:

```
class KNNModel(AbstractModel):
    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        self._model_type = self._get_model_type()

    def _get_model_type(self):
        if self.problem_type == REGRESSION:
            return KNeighborsRegressor
        else:
            return KNeighborsClassifier
```

Reply:
Ah, thanks - this makes a bit more sense than what I had in mind. Should be able to easily use both model types for comparisons now.